contracts/BookMarketplace.sol

```diff
@@ -44,15 +48,98 @@ contract BookMarketplace is ERC721URIStorage {
         bool sold
     );
 
+    event ListingUpdated(
+        uint256 indexed tokenId,
+        address seller,
+        address owner,
+        uint256 instantPrice,
+        uint256 startingPrice,
+        bool allowBid,
+        Bid[] bidList,
+        bool sold
+    );
+
+    /* modifiers */
+    
+    
+    // Modifier prevents function from being called by anyone other than the market owner
+    modifier onlyMarketOwner() {
+        require(
+            marketOwner == msg.sender,
+            "Only the marketplace owner can call this function."
+        );
+        _;
+    }
+
+    /* constructor */
+
     constructor() ERC721("BookExchange Tokens", "BOOK") {
-        owner = payable(msg.sender);
+        marketOwner = payable(msg.sender);
+    }
+
+    /* functions */
+
+    function getMarketOwnerAddress() public view returns(address) {
+        return marketOwner;
+    }
+
+    // change the price of a listing
+    function updateInstantPrice(uint256 tokenId, uint256 newPrice) public payable onlyMarketOwner {
+        Listing storage listing = idToMarketItem[tokenId];
+        listing.instantPrice = newPrice;
+        idToMarketItem[tokenId] = listing;
+    }
+
+    // get the price of a listing
+    function getInstantPrice(uint256 tokenId) public view returns (uint256) {
```

Comment:
While this is useful, do you think of a case where we only want to get the instant price instead of the whole listing?

Reply:
I wrote it for testing purposes mostly. For the main app, I imagine that we will likely retrieve full listings for the main app. Functions like this could be useful for more limited views of the data, but its most common use will definitely be in the test case.

Is there any value in reducing this functionality and making the tests a bit more complicated?

Reply:
I found value in taking it out, basically, the cost of deploying a smart contract goes up with the complexity of the contract. We don't really need to take it out now, but a concern that we should keep in mind (or report on) is that reduction of our smart contract complexity can cut down on the cost of deployment. 

However, gas cost for this wont really exist, as it is just a view.

Reply:
Great, as this is just a proof of theory, let keep it there. If we're going full production, then we're going to take this out.